### PR TITLE
Add installation flow for App Components

### PR DESCRIPTION
**Pull request by aw-asana**
Screen Shot 2022-01-18 at 9 36 01 PM
Screen Shot 2022-01-18 at 9 36 13 PM
Screen Shot 2022-01-18 at 9 36 22 PM
Screen Shot 2022-01-18 at 9 36 33 PM

### CODE REVIEW

File: source/includes/markdown/_13-platform-ui.html.md [hunk 1]
```diff
@@ -306,6 +306,77 @@ Before you begin, you'll need a developer sandbox in order to use App Components
 <br>
 That's it! At this point, feel free to keep exploring how changes in the server affects data in the task's Widget. Once you're ready to define an app, [click here to create your own app](/docs/configurations) with App Components.
 
+# Installing an App
+
+<span class="beta-indicator">BETA</span> - For access, please see [Overview of App Components](/docs/overview-of-app-components)
+
+When a user connects an app with App Components to Asana for the first time, they will go through an installation flow. This involves walking the user through the app's features (i.e., "value props"), authorizing the app, and adding it to projects in Asana. The installation flow can be triggered through either one of two ways:
```
Comment (ccankov) [inline source/includes/markdown/_13-platform-ui.html.md:313]: consider: do you think it's worth calling out explicitly that subsequent interactions with the same app by the same user won't trigger this installation flow?

consider: we have a deeplink to this installation flow that works regardless if a user has gone through it before. It looks like https://app.asana.com/-/install_platform_ui_app?app_id=<app_client_id> . Should we publicize this here?

File: source/includes/markdown/_13-platform-ui.html.md [hunk 2]
```diff
@@ -306,6 +306,77 @@ Before you begin, you'll need a developer sandbox in order to use App Components
...
+
+### Authenticating
+
+<span class="beta-indicator">BETA</span> - For access, please see [Overview of App Components](/docs/overview-of-app-components)
+
+On the next screen, the user will be directed to the auth screen, which will ask them to connect to the external app.
+
+<img src="../images/authentication.png" alt="authentication"/>
+
+When the user clicks the button to continue, they will go through an OAuth flow if the app has configured OAuth. If an [Asana OAuth](/docs/oauth) flow has been configured, it will appear first.
```
Comment (ccankov) [inline source/includes/markdown/_13-platform-ui.html.md:348]: Do: This is not really correct and we should likely update it. When the user clicks the button to continue, we make a request to the app's specified `authenticationUrl` in a popup window. From there, the developer chooses how to proceed with authentication. Usually this involves going through the Asana OAuth flow as well as a 3rd party OAuth flow (e.g. Jira, Zendesk, Zoom, etc), but technically this is all up to the developer's discretion - they can also include custom screens and logic that prompts the user to fill out additional information for setup of the app. The authentication flow is eventually concluded using the postMessage mechanism that is described elsewhere in the App Components docs.

**Review by ccankov — CHANGES_REQUESTED** [review]
Hitting request changes because I think it's important to accurately describe what happens as part of the authentication flow. Lmk if you want to chat through my comment on that!

File: source/includes/markdown/_13-platform-ui.html.md (hunk 1 above)
Comment (aw-asana) [reply source/includes/markdown/_13-platform-ui.html.md:313]: Definitely some helpful information (that would even benefit the QA process). I'll make sure it's in our content

File: source/includes/markdown/_13-platform-ui.html.md (hunk 2 above)
Comment (aw-asana) [reply source/includes/markdown/_13-platform-ui.html.md:348]: This is makes sense, and we should update the toolkit with this info as well. As far as the technical details of completing the authentication flow, I'll keep the link to https://developers.asana.com/docs/authorization at the end of this section.

**Review by ccankov — APPROVED** [review]
Looks great!! Thanks for following up :)

**Comment by aw-asana** [comment]
Thank you for the suggestions! Important stuff 👍

**Pull request merged**